packages/frontend/src/state/positions/hooks.ts

```diff
@@ -425,14 +425,34 @@ export const useVaultQuery = (vaultId: number) => {
   return { ...query, data: vaultData }
 }
 
+// check if there is no vault opened and can be reused
+// if there is an opened vault, then use the existing one
+// if there is no opened vault and had vaults in history then use the latest one to reopen a position
+// if there is no opened vault and had no vaults in history, then use 0 to open a new vault to create a position
 export const useFirstValidVault = () => {
-  const { vaults: shortVaults, loading } = useVaultManager()
-
-  const vault = shortVaults?.find((vault) => vault.collateralAmount.isGreaterThan(0))
-
+  const { vaults: shortVaults, loading, allVaults } = useVaultManager()
```

Comment:
Can we add tests for this logic with all the cases? The easiest way is to move the vault search algorithm to a function that accepts all the necessary hook variables as params and add tests for different conditions